_source/_posts/2022-08-29-react-typescript-redux.md

```diff
@@ -0,0 +1,530 @@
+tags: [javascript, typescript, react, redux, authentication]
+tweets:
+  - "Overwhelmed by state management options in #react? Learn about Redux, useState, and React Context and when to use each in this post by @gabzilladev! 🎉"
+image: blog/react-typescript-redux/social.jpg
+type: conversion
+github: https://github.com/oktadev/okta-react-typescript-redux-example
+---
+
+There are a myriad of state management options available for React. React provides the option of using the built-in [Context](https://reactjs.org/docs/context.html) for when you have a nested tree of components that share a state. There is also a built-in [useState hook](https://reactjs.org/docs/context.html) that will allow you to set local state for a component.
+
+For more complex scenarios where you might need a single source of truth that is shared across large sections of your application and is frequently changing, you might want to consider using a more robust state management library.
+
+This walkthrough will get you started with a basic setup using Okta's redirect model to manage authenticated state and user profile information within a React app. It provides examples on when to use [Redux](https://redux.js.org/), local state using React's `useState` hook, or React Context. 
+
+**Prerequisites**
+
+As of the time of this publication, Node >= 14.0.0 and npm >= 5.6 are required for Create React App. You can check for the latest required versions at [https://reactjs.org/docs/create-a-new-react-app.html](https://reactjs.org/docs/create-a-new-react-app.html).
+
+**NOTE:** Redux also recommends you install the [Chrome React DevTools Extension](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi?hl=en).
+{% include toc.md %}
+## Create the React app using TypeScript
+
+We'll get started with the built-in React template for [Create React App](https://github.com/facebook/create-react-app) that uses Redux and Typescript. This includes the recommended Redux Toolkit and Redux's integration with React components.
+
+1. First, run:
+```bash
+npx create-react-app okta-react-redux --template redux-typescript
+```
+
+**NOTE:** Our demo repository uses React `^18.2.0` and React Scripts `5.0.1`.
+
+2. Then we'll add the Redux core by running:
+```bash
+npm add redux@4.2
+```
+
+3. Redux provides its own types, but we'll want to add our `react-redux` types since we're using Typescript:
+```bash
+npm add -D @types/react-redux@7.1
+```
+
+4. We'll also need routing (and routing types) for our app:
+```bash
+npm install react-router-dom@5
+npm add -D @types/react-router-dom@5.3
+```
+
+5. To start your app, run:
+```bash
+npm start
+```
+
+## How Redux works
+
+To update state using Redux, an action is dispatched and the store then uses the root reducer to calculate a new state as compared to the old state, which then notifies the proper subscribers of the update so that the UI can properly be updated.
```

Comment:
```suggestion
To update state using Redux, an action is dispatched. Then the store uses the root reducer to calculate a new state as compared to the old state, which notifies the proper subscribers of the update, so that the UI can be updated properly.
```
Splitting a long sentence into two sentences for readability/comprehension